myproject.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:io'; // to add input,output
+
+void main() {
+  var noteApp = NoteTakingApp(); //Create an instance of the NoteTakingApp
+  var optione;
+
+  do {
+    noteApp.DisplayManu();
+    print("Enter a number from 1 to 5:");
+    optione = int.tryParse(stdin.readLineSync()!) ?? 0;
+
+    try {
+      switch (optione) {
+        // using swittch to evaluate the optione statment
+        case 1:
+          noteApp.Create();
+          break;
+
+        case 2:
+          noteApp.Edit();
+          break;
+
+        case 3:
+          noteApp.Delete();
+          break;
+
+        case 4:
+          noteApp.Search();
+          break;
+
+        case 5:
+          print("Exiting Note Taking App.");
+          print("---------------------------");
+          break;
+
+        default:
+          print("Invalid choice. please enter a number between 1 to 5.");
+          break;
+      }
+    } catch (e) {
+      print("An error acored: $e");
+    }
+  } while (optione != 5);
+}
+
+class NoteTakingApp {
+  var notes = <String, String>{}; //Map to store notes
+
+  void DisplayManu() {
+    print("\nNote Taking App");
+    print("1.Create a note");
+    print("2.Edit a note");
+    print("3.Delete a note");
+    print("4.Search for a note");
+    print("5.Exit the app");
+  }
+
+  void Create() {
+    print("Enter note title: ");
```

Comment:
check if the title already exists before creating a new note or if the title is empty when editing.